components/casemap/src/titlecase.rs

```diff
@@ -159,14 +159,14 @@ impl TitlecaseMapper<CaseMapper> {
             gc: Some(icu_properties::maps::general_category().static_to_owned()),
         }
     }
-    /// A constructor which creates a [`TitlecaseMapper`] using compiled data, with the legacy head adjustment behavior.
+    /// A constructor which creates a [`TitlecaseMapper`] using compiled data, with the "adjust to cased" head adjustment behavior.
     /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior and usage examples.
     ///
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
     #[cfg(feature = "compiled_data")]
-    pub const fn new_legacy() -> Self {
+    pub const fn new_adjust_to_cased() -> Self {
```

Comment:
Suggestion: on second thought, we don't really need this constructor. People can use `CaseMapper::titlecase_segment_adjust_to_cased` if they don't want to load the extra data.

Reply:
What if they want to still have it be the same type, though? Might make it clunky

Reply:
It's kind-of confusing that `HeadAdjustment` has 2 choices but there are 3 behaviors. We can add all 3 behaviors to that enum. Over in `CaseMapper::titlecase_segment_adjust_to_cased`, we decide what to do if given the enum variant which we don't support.

Reply:
yeah but the choice of behaviors affects data; we had kinda discussed this model and the 2x2 chart with the merged cells

Reply:
People who care about their data have a path; people who don't care as much about their data have a cleaner API. I think that's slightly better than a confusing API that affects everyone.

Reply:
Are you proposing we remove _specifically_ the compiled_data adjust_to_cased ctor? That seems okay.

Reply:
What should we do over FFI, then?